### PR TITLE
Stabilizing thermal turbojet thrust

### PR DESCRIPTION
**Pull request by dunclaw**
Ignore all changes except the one to ThermalNozzleController - I couldn't figure out GitHub UI for creating a PR for just one file.
When creating a spaceplane with a reactor and attached thermal turbojet I encountered wild oscillations in thrust at certain altitudes. I traced it to a feedback loop between thrust one frame and atmospheric intake the next frame. Adding a smoothing function between frames seems to fix it (and is reasonably realistic - the intake air won't wildly vary per frame).

### CODE REVIEW

File: GameData/WarpPlugin/Patches/NFTEnginesFix.cfg [hunk 1]
```diff
@@ -1,92 +1,3 @@
-@PART[ionXenon-*]:NEEDS[!NearFutureElectrical,!SETI,NearFuturePropulsion]:FOR[WarpPlugin]
```
Comment (sswelm) [inline GameData/WarpPlugin/Patches/NFTEnginesFix.cfg]: why remove?

File: FNPlugin/Collectors/AtmosphericIntake.cs [hunk 2]
```diff
@@ -61,6 +61,8 @@ public override void OnStart(PartModule.StartState state)
             jetTechBonusPercentage = 1 + (jetTechBonus / 10.736f);
         }
```
Comment (sswelm) [inline FNPlugin/Collectors/AtmosphericIntake.cs]: how would this work?

File: FNPlugin/Propulsion/ThermalNozzleController.cs [hunk 3]
```diff
@@ -1456,11 +1456,26 @@ public static double getIntakeAvailable(Vessel vess, string resourcename)
 
                 vess.rootPart.GetConnectedResourceTotals(resourceDefinition.id, out currentintakeatm, out maxintakeatm);
 
-                intake_amounts[resourcename] = currentintakeatm;
-                intake_maxamounts[resourcename] = maxintakeatm;
-            }
-
-            if (intake_amounts.ContainsKey(resourcename))
+				// to smooth out occilations in resource production, take a 90/10 weighted average of the previous frame production
```
Comment (sswelm) [inline FNPlugin/Propulsion/ThermalNozzleController.cs]: looks good

**Comment by sswelm** [comment]
Alright, so you say, I should only look at the ThermalNozzleController , which look like an improvement to me

File: FNPlugin/Collectors/AtmosphericIntake.cs (hunk 2 above)
Comment (dunclaw) [inline FNPlugin/Collectors/AtmosphericIntake.cs]: Ignore this - it is debug code where I'm trying to debug another issue. AtmosphericIntake seems to be drawing much less than it should. The reported flow in the context menu is much less than it actually receives. Haven't tracked that down yet.

File: GameData/WarpPlugin/Patches/NFTEnginesFix.cfg (hunk 1 above)
Comment (dunclaw) [inline GameData/WarpPlugin/Patches/NFTEnginesFix.cfg]: Just my preference - this patch breaks the ability to use NFT electrical engines with NFT solar panels which I sometimes use.

**Comment by sswelm** [comment]
Alright, I merged it manualy, thanks for you help. this will be closed

**Pull request closed without merging**